vertical-pod-autoscaler/pkg/admission-controller/logic/server.go

```diff
@@ -125,6 +96,53 @@ func (s *AdmissionServer) getPatchesForPodResourceRequest(raw []byte, namespace
 	return patches, nil
 }
 
+func getAddResourcesPatches(i int) patchRecord {
+	return patchRecord{
+		Op:    "add",
+		Path:  fmt.Sprintf("/spec/containers/%d/resources", i),
+		Value: v1.ResourceRequirements{},
+	}
+}
+
+func getAddResourcesSubPatch(i int, kind string) patchRecord {
+	return patchRecord{
+		Op:    "add",
+		Path:  fmt.Sprintf("/spec/containers/%d/resources/%s", i, kind),
+		Value: v1.ResourceList{},
+	}
+}
+
+func getAddResourceParamPatch(i int, kind, resource v1.ResourceName, quantity resource.Quantity) patchRecord {
```

Comment:
nit: Shouldn't kind be explicitly a string?

Reply:
Done.